org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/actions/RemoveAllCondtionalBreakpoints.java

```diff
@@ -0,0 +1,115 @@
+package org.eclipse.jdt.internal.debug.ui.actions;
+
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.resources.IWorkspaceRunnable;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.model.IBreakpoint;
+import org.eclipse.debug.internal.ui.DebugUIPlugin;
+import org.eclipse.debug.internal.ui.actions.breakpoints.RemoveAllTriggerPointsAction;
+import org.eclipse.jdt.debug.ui.IJavaDebugUIConstants;
+import org.eclipse.jdt.internal.debug.core.breakpoints.JavaLineBreakpoint;
+import org.eclipse.jdt.internal.debug.ui.JDIDebugUIPlugin;
+import org.eclipse.jface.action.IAction;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.MessageDialogWithToggle;
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.ui.IWorkbenchWindow;
+
+
+public class RemoveAllCondtionalBreakpoints extends RemoveAllTriggerPointsAction {
+	public RemoveAllCondtionalBreakpoints() {
+		super();
+	}
+	@Override
+	protected boolean isEnabled() {
+		for (IBreakpoint breakpoint : DebugPlugin.getDefault().getBreakpointManager().getBreakpoints()) {
+			if (breakpoint instanceof JavaLineBreakpoint javaBreakpoint) {
+				try {
+					if (javaBreakpoint.isConditionEnabled()) {
+						return true;
+					}
+				} catch (CoreException e) {
+					DebugUIPlugin.log(e);
+				}
+			}
+		}
+		return false;
+	}
+
+	@Override
+	public void run(IAction action) {
+		IWorkbenchWindow window = JDIDebugUIPlugin.getActiveWorkbenchWindow();
+		if (window == null) {
+			return;
+		}
+		IPreferenceStore store = JDIDebugUIPlugin.getDefault().getPreferenceStore();
+		boolean prompt = store.getBoolean(IJavaDebugUIConstants.PREF_PROMPT_REMOVE_ALL_CONDITIONAL_BREAKPOINTS);
+		boolean proceed = true;
+		if (prompt) {
+			MessageDialogWithToggle mdwt = MessageDialogWithToggle.openYesNoQuestion(window.getShell(), ActionMessages.ClearAllCondtionsInBreakpointsAction_0, ActionMessages.ClearAllCondtionsInBreakpointsAction_1, ActionMessages.ClearAllCondtionsInBreakpointsAction_2, !prompt, null, null);
+			if (mdwt.getReturnCode() != IDialogConstants.YES_ID) {
+				proceed = false;
+			} else {
+				store.setValue(IJavaDebugUIConstants.PREF_PROMPT_REMOVE_ALL_CONDITIONAL_BREAKPOINTS, !mdwt.getToggleState());
+			}
+		}
+		if (proceed) {
+			new Job(ActionMessages.ClearAllCondtionsInBreakpointsAction_1) {
+				@Override
+				protected IStatus run(IProgressMonitor monitor) {
+					try {
+						for (IBreakpoint breakpoint : DebugPlugin.getDefault().getBreakpointManager().getBreakpoints()) {
+							if (breakpoint instanceof JavaLineBreakpoint javaBp) {
+								if (javaBp.isConditionEnabled()) {
+									javaBp.setConditionEnabled(false);
```

Comment:
I'm not sure this is the right thing to do. I would expect the breakpoint to be disabled, not the condition on the breakpoint

Reply:
Intention of the PR is to provide functionality similar to "Remove All Triggers" in which all triggers in breakpoints will be disabled by preserving the breakpoint enabled/disabled status.

Reply:
so for this all conditional check will be disabled. keeping the condition and enabled/disabled status as it is.